es6-shim.js

```diff
@@ -1387,7 +1387,8 @@
     var isBadHex = isBadHexRegex.test.bind(isBadHexRegex);
     var NumberShim = (function () {
       // this is wrapped in an IIFE because of IE 6-8's wacky scoping issues with named function expressions.
-      var NumberShim = function Number(value) {
+      var NumberShim = function Number() {
+        var value = arguments[0];
```

Comment:
but i'm fine using the *value* out of the arguments object - however, if `arguments.length < 1`, `arguments[0]` deoptimizes in many engines, including older v8.